### PR TITLE
Conversion to webextension format

### PR DESCRIPTION
**Pull request by Schrikvis**
I tried my best to keep your toolchain working, but frankly I've probably messed everything up entirely. 😔

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -1,27 +0,0 @@
-{
```
Comment (Rudloff) [inline package.json]: Is there any particular reason to delete package.json? We still need it in order to install the Grunt tasks.

File: data/js/emoji.js [hunk 2]
```diff
@@ -1,5 +1,5 @@
-/*global twemoji, self*/
-/*jslint browser: true, devel: true*/
+-/*global twemoji*/
+-/*jslint browser: true, devel: true*/
```
Comment (Rudloff) [inline data/js/emoji.js]: You have some buggy hyphens at the start of the lines here.

File: data/js/twemoji.min.js [hunk 3]
```diff
@@ -0,0 +1,2 @@
+/*! Copyright Twitter Inc. and other contributors. Licensed under MIT */
```
Comment (Rudloff) [inline data/js/twemoji.min.js]: Please don't include `twemoji.min.js` in the repository. It is cleaner to install it with Bower.

File: manifest.json [hunk 4]
```diff
@@ -0,0 +1,44 @@
...
+  "description": "Convert unicode emoji to images on all webpages",
+  "version": "0.1.5-beta",
+  "author": "Pierre Rudloff",
+  "developer": {
+    "name": "Pierre Rudloff <contact@rudloff.pro>"
+  },
+  "incognito": "spanning",
+  "bugs": "https://github.com/Rudloff/emoji-everywhere/issues",
+  "homepage_url": "https://addons.mozilla.org/firefox/addon/emoji-everywhere/",
+  "devDependencies": {
```
Comment (Rudloff) [inline manifest.json]: We don't need the following properties in `manifest.json`:
* `bugs`
* `devDependencies`
* `keywords`
* `license`
* `main`
* `repository`

File: data/js/emoji.js [hunk 5]
```diff
@@ -1,5 +1,5 @@
...
 if (twemoji === undefined) {
     console.error("Can't load Twemoji");
 } else {
```
Comment (Rudloff) [inline data/js/emoji.js]: `self` does not exist anymore in webextensions. But we can do something like this:
```js
twemoji.base = browser.extension.getURL('data/components/twemoji/2/');
```

File: manifest.json [hunk 6]
```diff
@@ -0,0 +1,44 @@
...
+  "content_scripts": [
+    {
+      "matches": ["<all_urls>"],
+      "js": [
+	    "data/js/twemoji.min.js",
+	    "data/js/emoji.js"
+	  ],
+	  "css": ["data/css/emoji.css"]
+    }
+  ],
```
Comment (Rudloff) [inline manifest.json:25]: We will also need something like this in order to allow the add-on to load the images:
```json
"web_accessible_resources": [
  "data/components/twemoji/2/72x72/*.png"
],
```

File: manifest.json [hunk 7]
```diff
@@ -0,0 +1,44 @@
...
+  "icons": {
+    "48": "icons/emoji-48.png",
+    "96": "icons/emoji-96.png"
+  },
+  "content_scripts": [
+    {
+      "matches": ["<all_urls>"],
+      "js": [
+	    "data/js/twemoji.min.js",
```
Comment (Rudloff) [inline manifest.json]: Please use the one installed by Bower in `data/components/twemoji/2/twemoji.min.js`.

**Review by Rudloff — CHANGES_REQUESTED** [review]
Thanks a lot, that's a great start!
I added a few comments to your changes. Some are required to make the add-on work correctly and others are just some good practices stuff.

Please tell me if you don't feel confortable doing some of the changes, I can do them myself if you prefer, since you already did most of the work.

File: package.json (hunk 1 above)
Comment (Schrikvis) [reply package.json]: I was worried it'd confuse firefox about the nature of the extension, but that's nonsense and it's back now, sorry.

**Comment by Schrikvis** [comment]
Take two, I think I've made all the necessary changes.

**Review by Rudloff — CHANGES_REQUESTED** [review]
Thanks again, everything looks good!
Just one final little change and then I will merge: now that package.json is back, can you please restore the fixpack task in Gruntfile.js?

**Review by Rudloff — APPROVED** [review]

**Comment by Schrikvis** [comment]
❤

**Pull request merged**